src/value/owned.rs

```diff
@@ -464,6 +467,39 @@ impl OwnedElement {
     }
 }
 
+impl Display for OwnedElement {
+    fn fmt(&self, f: &mut Formatter<'_>) -> Result<(), std::fmt::Error> {
+        let mut ivf = IonValueFormatter {
+            output: &mut Vec::new(),
+            string_escape_codes: string_escape_code_init(),
+        };
```

Comment:
Ah, shoot. I hadn't thought about this bit. 😞 

We don't want to have to allocate a new `Vec` and buffer all the output each time someone uses `Display` -- we'd like to be able to write directly to `f`. Unfortunately, I think this means we have to use `fmt::Write` throughout `IonValueFormatter` instead of `io::Write`. (If we genuinely need to format text to an `io::Write` in some other context, I can think of a way to achieve that.)

Let me know if you'd like to discuss this offline. The `io::Write`/`fmt::Write` divide is definitely one of the stranger parts of the standard library that I've encountered. The